src/services/sync-component-with-file/unlink-component.ts

```diff
@@ -18,7 +35,7 @@ export const unlinkComponent = async (componentId: string, repoId: string): Prom
 
   try {
     await deleteExternalAlias({
```

Comment:
We should follow all-or-nothing pattern here, (if detach is successful and removing external alias call fails)
We have decided to publish metrics to SFX for the following:
1. Success/Failure of the detachDataManager calls
2. Success/Failure of the removeExternalAliases calls

If the Failure rate of detachDataManager is high, then we will build an Application level rollback mechanism, which will re-attach the data manager if the removeExternalAliases fails.

I have added a comment on https://softwareteams.atlassian.net/browse/COMPASS-12687 w.r.t this.